standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/utils/SecurityUtils.java

```diff
@@ -271,6 +286,31 @@ public static TTransport getSSLSocket(String host, int port, int loginTimeout,
     return getSSLSocketWithHttps(tSSLSocket);
   }
 
+  public static THttpClient getHttpSSLSocket(String httpsUrl, String trustStorePath,
+      String trustStorePasswd, String trustStoreAlgorithm, String trustStoreType)
```

Comment:
I did not find http server related code, for example, the server ssl  socket that can handle http request. Will the servlet handle that?

Reply:
The server side code for this is in HiveMetaStore.java at https://github.com/apache/hive/pull/3105/files#diff-00e70b6958060aa36762b21bf16676f83af01c1e09b56816aecc6abe7c8ac866R683